theano/tensor/tests/test_basic.py

```diff
@@ -1932,6 +1932,61 @@ def _numpy_true_div(x, y):
                         bad_runtime=dict(bad1=(rand(5, 7), rand(5, 7)),
                                          bad2=(rand(5, 7), rand(8, 3))))
 
+BatchedDotTester = makeTester(
+    name='BatchedDotTester',
+    op=batched_dot,
+    expected=(lambda xs, ys:
+        numpy.asarray(
+            list(x * y if x.ndim == 0 or y.ndim == 0
+                 else (numpy.dot(x, y) if y.ndim == 1
```

Comment:
In fact, I would do 

if x.ndim == 0 or y.ndim == 0:
 ...
else: numpy.dot(x, y)

the call to tensordot bellow could be buggy and is equivalent in this case. So this make simpler test.